nixos/modules/system/boot/stage-1-init.sh

```diff
@@ -119,6 +119,18 @@ specialMount() {
 }
 source @earlyMountScript@
 
+# Copy initrd secrets from /.initrd-secrets to their actual destinations
+if [ -d "/.initrd-secrets" ]; then
+    #
+    # Secrets are named by their full destination pathname and stored
+    # under /.initrd-secrets/
+    #
+    for secret in $(cd "/.initrd-secrets"; find . -type f); do
```

Comment:
```suggestion
    for secret in $(find /.initrd-secrets -not -type d); do
```

Reply:
Hello! @SuperSandro2000 [suggested the same thing here](https://github.com/NixOS/nixpkgs/pull/129449#discussion_r664689775), but that changes the output and paths assigned to `secret` (they would have `/.initrd-secrets/` still prepended vs. being the full path specified as the attribute name to `boot.initrd.secrets`. I chose this approach rather than having to strip the first path component inside the loop, which would be less precise. In addition, `-not -type d` is not equivalent to explicitly including files (`-type f`), it could include links, sockets, device nodes, etc. 

I tried to explain the above in the comment preceding the loop, but if it's still not clear to reviewers, I should try to clarify more.